synthesizer/process/src/stack/mod.rs

```diff
@@ -175,8 +175,8 @@ impl<N: Network> CallStack<N> {
 pub struct Stack<N: Network> {
     /// The program (record types, structs, functions).
     program: Program<N>,
-    /// The mapping of external stacks as `(program ID, stack)`.
-    external_stacks: IndexMap<ProgramID<N>, Arc<Stack<N>>>,
+    /// A reference to the global stack map.
+    stacks: Weak<RwLock<IndexMap<ProgramID<N>, Arc<Stack<N>>>>>,
```

Comment:
Is it possible for the Stack not to hold any reference to the global set of Stacks, and instead pass in relevant imports for the respective functions which need them (notably `initialize`)? Might be a cleaner design, and at least in the context of storing stacks to disk it would have made things easier - not sure if this still holds after program upgradability.

Reply:
I explored this idea initially, but it involves a larger refactor. 
Basically, you'd end up passing `Process` around wherever you call `get_external_stack`. 
Some places like `initialize` are straightforward, however you get into really hairy territory when you get to `call.rs`, where the stack is trait-ified.